packages/angular/projects/clr-angular/src/forms/common/if-control-state/if-error.ts

```diff
@@ -28,17 +28,20 @@ export class ClrIfError extends AbstractIfState {
    * @param state CONTROL_STATE
    */
   protected handleState(state: CONTROL_STATE) {
-    const isInvalid = CONTROL_STATE.INVALID === state;
+    if (this.error && this.control) {
+      this.displayError(this.control.hasError(this.error));
+    } else {
+      this.displayError(CONTROL_STATE.INVALID === state);
+    }
+  }
 
-    if (isInvalid && this.displayedContent === false) {
-      let options = {};
-      if (this.error && this.control && this.control.hasError(this.error)) {
-        options = { error: this.control.getError(this.error) };
-      }
-      this.container.createEmbeddedView(this.template, options);
-    } else if (!isInvalid && this.container) {
+  private displayError(invalid: boolean) {
+    if (invalid && this.displayedContent === false) {
```

Comment:
Do we need to do a check for the container before trying to create a view?

Reply:
I don't think so. At least I could not trigger an error for that.

In the case above, `this.container` is the `host` container so it exists.
In the previews implementation, the `displayError` was call in the constructor stage so there may be cases where the `host` is not ready. But this is not the case anymore.

I try a few different combinations and did not manage to trigger the `TypeError: Cannot read property`.
From what I suspect the `else if` check for the container is for cases when the directive is destroyed and at the same time a `displayError` is called (because it's a callback and could be called async) to prevent an unnecessary code execution. 

I could put the additional check to be double sure?

Reply:
I think it would be useful to bail out if container is not present, just to avoid lifecycle issues and simplify the logic.